### PR TITLE
Add flag to publish intermediate bracken files

### PR DESCRIPTION
**Pull request by sofstam**
Fixes the bug reported [here](https://nfcore.slack.com/archives/C031QH57DSS/p1716196030609319?thread_ts=1715135004.568159&cid=C031QH57DSS)

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/modules.config [hunk 1]
```diff
@@ -467,7 +467,8 @@ process {
         publishDir = [
             path: { "${params.outdir}/kraken2/${meta.db_name}/" },
             mode: params.publish_dir_mode,
-            pattern: '*.{txt,fastq.gz}'
+            pattern: '*.{txt,fastq.gz}',
+            saveAs: {  params.bracken_save_intermediatekraken2 == false && meta.tool == bracken ? false : true  }
```
Comment (jfy133) [inline conf/modules.config:471]:
```suggestion
            saveAs: {  params.bracken_save_intermediatekraken2 == false && meta.tool == bracken ? null : it  }
```

Sorry, forgot about that 😅. The true/false applies when it's under `enabled` save has to be a name.

At least I hope so, comparing to the adapterremoval thing above

**Comment by sofstam** [comment]
Realised that yesterday looking at the docs of publishDir. Still not working, I found another example and maybe we can provide the condition in the path.

File: docs/output.md [hunk 2]
```diff
@@ -389,6 +389,8 @@ The main taxonomic classification file from Kraken2 is the `_combined_reports.tx
 
 You will only receive the `.fastq` and `*classifiedreads.txt` file if you supply `--kraken2_save_reads` and/or `--kraken2_save_readclassifications` parameters to the pipeline.
 
+When running Kraken2 together with bracken, you can use the `--bracken_save_intermediatekraken2` to save the intermediate/upstream files generated during the bracken run.
```
Comment (jfy133) [inline docs/output.md:392]:
```suggestion
When running Bracken, you will only get the 'intermediate' Kraken2 report files in this directory if you supply `--bracken_save_intermediatekraken2` to the run.
```

Comment (jfy133) [reply docs/output.md:392]: Please also add  on lines 379 and 383 above that the Bracken files only with also with the flag too

File: nextflow_schema.json [hunk 3]
```diff
@@ -466,6 +466,11 @@
                     "description": "Turn on Bracken (and the required Kraken2 prerequisite step).",
                     "fa_icon": "fas fa-toggle-on"
                 },
+                "bracken_save_intermediatekraken2": {
+                    "type": "boolean",
+                    "fa_icon": "fas fa-save",
+                    "description": "Turn on saving the upstream Bracken files"
```
Comment (jfy133) [inline nextflow_schema.json:472]:
```suggestion
                    "description": "Turn on the saving of the intermediate Kraken2 files used as input to Bracken itself into Kraken2 results folder"
```

**Review by jfy133 — COMMENTED** [review]
Need a changelog update (you can can add Emily Wissel as the reporter)

File: CHANGELOG.md [hunk 4]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### `Fixed`
 
 - [#484](https://github.com/nf-core/taxprofiler/pull/484) Improved input validation to immediately fail if run accession IDs within a given sample ID are not unique (❤️ to @sofstam for reporting, fixed by @jfy133)
+- [491](https://github.com/nf-core/taxprofiler/pull/491) Added flag to publish intermediate bracken files (❤️ to @ewissel for reporting, fixed by @sofstam and @jfy133)
```
Comment (jfy133) [inline CHANGELOG.md:15]:
```suggestion
- [#491](https://github.com/nf-core/taxprofiler/pull/491) Added flag to publish intermediate bracken files (❤️ to @ewissel for reporting, fixed by @sofstam and @jfy133)
```

**Review by jfy133 — APPROVED** [review]

**Pull request merged**